be/src/exec/vectorized/schema_scan_node.cpp

```diff
@@ -0,0 +1,254 @@
+    }
+
+    if (tnode.schema_scan_node.__isset.current_user_ident) {
+        _scanner_param.current_user_ident = _pool->add(new TUserIdentity(tnode.schema_scan_node.current_user_ident));
+    } else {
+        if (tnode.schema_scan_node.__isset.user) {
+            _scanner_param.user = _pool->add(new std::string(tnode.schema_scan_node.user));
+        }
+        if (tnode.schema_scan_node.__isset.user_ip) {
+            _scanner_param.user_ip = _pool->add(new std::string(tnode.schema_scan_node.user_ip));
+        }
+    }
+
+    if (tnode.schema_scan_node.__isset.ip) {
+        _scanner_param.ip = _pool->add(new std::string(tnode.schema_scan_node.ip));
+    }
+    if (tnode.schema_scan_node.__isset.port) {
+        _scanner_param.port = tnode.schema_scan_node.port;
+    }
+
+    if (tnode.schema_scan_node.__isset.thread_id) {
+        _scanner_param.thread_id = tnode.schema_scan_node.thread_id;
+    }
+    return Status::OK();
+}
+
+Status SchemaScanNode::prepare(RuntimeState* state) {
+    if (_is_init) {
+        return Status::OK();
+    }
+
+    if (NULL == state) {
+        return Status::InternalError("input pointer is NULL.");
+    }
+
+    RETURN_IF_ERROR(ScanNode::prepare(state));
+
+    // get dest tuple desc
+    _dest_tuple_desc = state->desc_tbl().get_tuple_descriptor(_tuple_id);
+
+    if (NULL == _dest_tuple_desc) {
+        return Status::InternalError("Failed to get tuple descriptor.");
+    }
+
+    const SchemaTableDescriptor* schema_table =
+            static_cast<const SchemaTableDescriptor*>(_dest_tuple_desc->table_desc());
+
+    if (NULL == schema_table) {
+        return Status::InternalError("Failed to get schema table descriptor.");
+    }
+
+    // new one scanner
+    _schema_scanner.reset(SchemaScanner::create(schema_table->schema_table_type()));
+
+    if (NULL == _schema_scanner.get()) {
+        return Status::InternalError("schema scanner get NULL pointer.");
+    }
+
+    _schema_scanner->set_slot_descs(_dest_tuple_desc->slots());
+
+    RETURN_IF_ERROR(_schema_scanner->init(&_scanner_param, _pool));
+
+    _is_init = true;
+
+    return Status::OK();
+}
+
+Status SchemaScanNode::open(RuntimeState* state) {
+    if (!_is_init) {
+        return Status::InternalError("Open before Init.");
+    }
+
+    if (NULL == state) {
+        return Status::InternalError("input pointer is NULL.");
+    }
+
+    SCOPED_TIMER(_runtime_profile->total_time_counter());
+    RETURN_IF_ERROR(exec_debug_action(TExecNodePhase::OPEN));
+    RETURN_IF_CANCELLED(state);
+    RETURN_IF_ERROR(ExecNode::open(state));
+
+    if (_scanner_param.user) {
+        TSetSessionParams param;
+        param.__set_user(*_scanner_param.user);
+        //TStatus t_status;
+        //RETURN_IF_ERROR(SchemaJniHelper::set_session(param, &t_status));
+        //RETURN_IF_ERROR(Status(t_status));
+    }
+
+    return _schema_scanner->start(state);
+}
+
+Status SchemaScanNode::get_next(RuntimeState* state, RowBatch* row_batch, bool* eos) {
```

Comment:
Don't need the intferface

Reply:
ExecNode does not implement one, we need a default one here.